### PR TITLE
[JENKINS-26100] Change buildEnvVars to take Run

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-26100](https://issues.jenkins-ci.org/browse/JENKINS-26100)

FYI, I had some weird test failures locally when I ran `mvn clean install` but couldn't reproduce when running, say, `mvn clean install -Dtest=PipelineTest` so I'm not sure if they're environmental or what, so...hey, why not, here's a PR.

cc @reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,9 +14,10 @@
     <url>http://wiki.jenkins-ci.org/display/JENKINS/Mercurial+Plugin</url>
     
     <properties>
-      <jenkins.version>1.642.3</jenkins.version>
+        <jenkins-core.version>2.58-20170501.171905-3</jenkins-core.version> <!-- TODO: Switch to release once https://github.com/jenkinsci/jenkins/pull/2730 is merged and released -->
+        <jenkins-war.version>2.58-20170501.171927-3</jenkins-war.version> <!-- TODO: Switch to release once https://github.com/jenkinsci/jenkins/pull/2730 is merged and released -->
       <no-test-jar>false</no-test-jar>  
-      <scm-api-plugin.version>2.0.8</scm-api-plugin.version>
+      <scm-api-plugin.version>2.1.0</scm-api-plugin.version>
```
Comment (jglick) [inline pom.xml]: `java.level` → 8

File: pom.xml [hunk 2]
```diff
@@ -14,9 +14,10 @@
     <url>http://wiki.jenkins-ci.org/display/JENKINS/Mercurial+Plugin</url>
     
     <properties>
-      <jenkins.version>1.642.3</jenkins.version>
+        <jenkins-core.version>2.58-20170502.192524-8</jenkins-core.version> <!-- TODO: Switch to release once https://github.com/jenkinsci/jenkins/pull/2730 is merged and released -->
+        <jenkins-war.version>2.58-20170502.192544-8</jenkins-war.version> <!-- TODO: Switch to release once https://github.com/jenkinsci/jenkins/pull/2730 is merged and released -->
```
Comment (jglick) [inline pom.xml]: 2.61

Comment (jglick) [inline pom.xml]: 2.60 I meant.

File: src/main/java/hudson/plugins/mercurial/MercurialSCM.java [hunk 3]
```diff
@@ -881,7 +881,11 @@ private void clone(Run<?, ?> build, Launcher launcher, FilePath repository, Node
     }
 
     @Override
-    public void buildEnvVars(AbstractBuild<?,?> build, Map<String, String> env) {
+    public void buildEnvVars(AbstractBuild<?,?> build, Map<String,String> env) {
+        buildEnvironment(build, env);
+    }
+
+    public void buildEnvironment(Run<?,?> build, Map<String, String> env) {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/mercurial/MercurialSCM.java]: Same as for SVN plugin. It is better to reference the incoming method in the Core API.

File: pom.xml [hunk 4]
```diff
@@ -16,7 +16,7 @@
     <properties>
       <jenkins.version>1.642.3</jenkins.version>
       <no-test-jar>false</no-test-jar>  
-      <scm-api-plugin.version>2.0.8</scm-api-plugin.version>
+      <scm-api-plugin.version>2.1.0</scm-api-plugin.version>
```
Comment (oleg-nenashev) [inline pom.xml]: I doubt it is required, but it won't hurt

File: src/main/java/hudson/plugins/mercurial/MercurialSCM.java (hunk 3 above)
Comment (abayer) [inline src/main/java/hudson/plugins/mercurial/MercurialSCM.java]: Isn't that literally what I just did here?

Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/mercurial/MercurialSCM.java]: Not sure WDYM. I was talking about Javadoc and TODO comment

Comment (abayer) [inline src/main/java/hudson/plugins/mercurial/MercurialSCM.java]: Ah, yeah, doing so.

**Pull request merged**